packages/babel-parser/test/fixtures/flow/typecasts/fail-without-parens-jsx-tokens-true/input.js

```diff
@@ -0,0 +1 @@
+<div propA={[key: value]} />
```

Comment:
Currently the `tokens` of [parsed AST](https://astexplorer.net/#/gist/9675cd0445add4e9c24d321fc433be7a/1f4d63fd000b655fedfe1585131fac08b614e4fd) is a length-14 array
```js
[ tt.jsxTagStart, tt.name, undefined, undefined, ... , tt.eof]
```

`babel7Compat` thus throws because the `tokens` unexpectedly includes `undefined`.